source/uwp/Renderer/lib/WholeItemsPanel.cpp

```diff
@@ -519,4 +683,64 @@ AdaptiveNamespaceStart
 
         return !isnan(definedImageHeight) || !isnan(definedImageWidth);
     }
+
+    bool WholeItemsPanel::GridIsAColumnSet(_In_ IGrid* grid)
+    {
+        bool gridIsColumnSet{true};
+
+        ComPtr<IGrid> localGrid(grid);
+        ComPtr<IVector<RowDefinition*>> rowDefinitions;
+        THROW_IF_FAILED(localGrid->get_RowDefinitions(&rowDefinitions));
+        unsigned int rowCount{};
+        THROW_IF_FAILED(rowDefinitions->get_Size(&rowCount));
+
+        if (rowCount <= 1)
+        {
+            ComPtr<IPanel> gridAsPanel;
+            THROW_IF_FAILED(localGrid.As(&gridAsPanel));
+            ComPtr<IVector<UIElement*>> columns;
+            THROW_IF_FAILED(gridAsPanel->get_Children(columns.ReleaseAndGetAddressOf()));
+            unsigned int columnCount{};
+            THROW_IF_FAILED(columns->get_Size(&columnCount));
+
+            for (unsigned int i{}; i < columnCount && gridIsColumnSet; ++i)
+            {
+                ComPtr<IUIElement> column;
+                THROW_IF_FAILED(columns->GetAt(i, column.GetAddressOf()));
+
+                // If the child is a WholeItemsPanel it's ok
+                ComPtr<IWholeItemsPanel> columnAsWholeItemsPanel;
+                if (FAILED(column.As(&columnAsWholeItemsPanel)))
+                {
+                    // If the child is not a WholeItemsPanel, it may be a separator
+                    ComPtr<IGrid> columnAsGrid;
+                    if (SUCCEEDED(column.As(&columnAsGrid)))
+                    {
+                        ComPtr<IPanel> separatorAsPanel;
+                        THROW_IF_FAILED(columnAsGrid.As(&separatorAsPanel));
+                        ComPtr<IVector<UIElement*>> separatorContents;
+                        THROW_IF_FAILED(separatorAsPanel->get_Children(separatorContents.ReleaseAndGetAddressOf()));
+                        unsigned int separatorContentsCount{};
+                        THROW_IF_FAILED(separatorContents->get_Size(&separatorContentsCount));
+
+                        if (separatorContentsCount != 0)
+                        {
+                            gridIsColumnSet = false;
```

Comment:
>gridIsColumnSet = false; [](start = 28, length = 24)

break here so we stop looping

Reply:
the loop should stop after that value is set as the loop condition is
i < columnCount && gridIsColumnSet

---
In reply to: [183096931](https://github.com/Microsoft/AdaptiveCards/pull/1377#discussion_r183096931) [](ancestors = 183096931)

Reply:
Indeed, missed that.

---
In reply to: [183479578](https://github.com/Microsoft/AdaptiveCards/pull/1377#discussion_r183479578) [](ancestors = 183479578,183096931)